modules/foreman_ptable.py

```diff
@@ -0,0 +1,335 @@
+# Alternatively make use of with_filetree to parse recursively with filter.
+- name: "Parsing a directory of partition templates"
+  local_action:
+      module: foreman_ptable
+      username: "admin"
+      password: "admin"
+      server_url: "https://foreman.example.com"
+      file_name: "{{ item }}"
+      state: present
+      locations:
+      - SKARO
+      organizations:
+      - DALEK INC
+      with_fileglob:
+       - "./arsenal_templates/*.erb"
+
+'''
+
+RETURN = ''' # '''
+try:
+    from nailgun.entities import (
+        PartitionTable,
+        OperatingSystem,
+        _OPERATING_SYSTEMS,
+    )
+
+    from ansible.module_utils.ansible_nailgun_cement import (
+        Organization,
+        Location,
+        create_server,
+        find_entity,
+        find_entities,
+        create_entity,
+        update_entity,
+        parse_template,
+        parse_template_from_file,
+    )
+    HAS_NAILGUN_PACKAGE = True
+except ImportError:
+    HAS_NAILGUN_PACKAGE = False
+
+import os
+from ansible.module_utils.basic import AnsibleModule, get_module_path
+from ansible.module_utils._text import to_bytes, to_native
+
+
+def sanitize_ptable_dict(ptable_dict):
+    # This is the only true source for names (and conversions thereof)
+    name_map = {
+        'layout': 'layout',
+        'locations': 'location',
+        'name': 'name',
+        'organizations': 'organization',
+        'os_family': 'os_family',
+    }
+    result = {}
+    for key, value in name_map.iteritems():
+        if key in ptable_dict:
+            result[value] = ptable_dict[key]
+    return result
+    # Missing parameters:
+    # snippet
+    # locked
+    # audit_comment
+    # default
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True),
+            username=dict(required=True, no_log=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(required=False, type='bool', default=True),
+            # audit_comment=dict(required=False),
+            layout=dict(required=False),
+            file_name=dict(required=False, type='path'),
+            locations=dict(required=False, type='list'),
+            # locked=dict(required=False, type='bool', default=False),
+            name=dict(required=False),
+            organizations=dict(required=False, type='list'),
+            os_family=dict(required=False, choices=list(_OPERATING_SYSTEMS)),
+            state=dict(required=True, choices=['absent', 'present', 'latest']),
+        ),
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ['file_name', 'layout'],
+        ],
+        required_one_of=[
+            ['file_name', 'layout'],
+        ],
+
+    )
+    if not HAS_NAILGUN_PACKAGE:
+        module.fail_json(
+            msg='Missing required nailgun module'
+                '(check docs or install with: pip install nailgun')
+
+    changed = False
```

Comment:
The logic implemented here (absent, present, latest) is somewhat more complicated, however we contracted it into the library.
If you have a better function name than 'ansity' let us know.